server/src/main/java/com/gitittogether/skillForge/server/config/JwtAuthenticationFilter.java

```
@@ -0,0 +1,64 @@
+package com.gitittogether.skillForge.server.config;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+/**
+ * This filter:
+ * 1) Reads a JWT from a named cookie (e.g. "AuthToken").
+ * 2) Validates the token.
+ * 3) If valid, sets an authenticated Principal in SecurityContext.
+ */
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private final JwtUtils jwtUtils;
+    private final JwtUserDetailsService userDetailsService; // See below
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    @NonNull HttpServletResponse response,
+                                    @NonNull FilterChain filterChain) throws ServletException, IOException {
+        String authHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+        String jwt = null;
+        String userId = null;
+
+        // JWT should be in the form "Bearer eyJ..."
+        if (authHeader != null && authHeader.startsWith("Bearer ")) {
+            jwt = authHeader.substring(7);
+            try {
+                userId = jwtUtils.extractUserId(jwt);
+            } catch (Exception ignored) {
```

The exception handling is silent (empty catch block), which could make debugging difficult. Consider at least logging the exception or implementing a more robust error handling strategy.